### PR TITLE
Table: Optimization - render icons on hover

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**

Only rendering icons on hover

**Why do we need this feature?**
Table component has too many DOM nodes, removing the SVG icons from being rendered in each component speeds up render significantly.

**Who is this feature for?**
Users of Table component

**Special notes for your reviewer:**
Table cells don't seem to get keyboard focus, once they do this will need to be changed to include onFocus/onBlur

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Table/DefaultCell.tsx [hunk 1]
```diff
@@ -41,7 +49,7 @@ export const DefaultCell = (props: TableCellProps) => {
   }
 
   return (
-    <div {...cellProps} className={cellStyle}>
+    <div {...cellProps} onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave} className={cellStyle}>
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/Table/DefaultCell.tsx:52]: you should skip this if `showActions` is false

Comment (gtk-grafana) [reply packages/grafana-ui/src/components/Table/DefaultCell.tsx]: Done

File: packages/grafana-ui/src/components/Table/DefaultCell.tsx [hunk 2]
```diff
@@ -41,7 +49,12 @@ export const DefaultCell = (props: TableCellProps) => {
...
 
   return (
-    <div {...cellProps} className={cellStyle}>
+    <div
+      {...cellProps}
+      onMouseEnter={showActions ? onMouseEnter : undefined}
+      onMouseLeave={showActions ? onMouseLeave : undefined}
+      className={cellStyle}
+    >
       {!hasLinks && <div className={tableStyles.cellText}>{value}</div>}
```
Comment (gtk-grafana) [inline packages/grafana-ui/src/components/Table/DefaultCell.tsx:58]: We also tried to remove this div as well, but there are some viz within table cell that require the cell to be display:flex that won't work with text-overflow:ellipsis

Comment (gtk-grafana) [reply packages/grafana-ui/src/components/Table/DefaultCell.tsx:58]: Addressed in https://github.com/grafana/grafana/pull/76916

**Review by codeincarnate — APPROVED** [review]
Tested this with a larger table and even without profiling it visibly improves rendering performance 🙌 . Couldn't find any issues in terms of styling or such. LGTM

**Comment by gtk-grafana** [comment]
Huzzah!

File: packages/grafana-ui/src/components/Table/styles.ts [hunk 3]
```diff
@@ -11,14 +11,30 @@ export function useTableStyles(theme: GrafanaTheme2, cellHeightOption: TableCell
...
+  ) => {
     return css({
       label: overflowOnHover ? 'cellContainerOverflow' : 'cellContainerNoOverflow',
       padding: `${cellPadding}px`,
       width: '100%',
       // Cell height need to account for row border
       height: `${rowHeight - 1}px`,
-      display: 'flex',
+
+      display: asCellText ? 'block' : 'flex',
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Table/styles.ts:27]: @gtk-grafana This breaks the filter actions that now appear below the row/cell, making the filter action tooltip block out the cell content. 

Screenshot 2024-02-07 at 15 58 38

**Pull request merged**